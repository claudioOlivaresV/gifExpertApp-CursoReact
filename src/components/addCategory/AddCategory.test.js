import TextField from '@material-ui/core/TextField';
import '@testing-library/jest-dom'
const { shallow } = require("enzyme");
const { AddCategory } = require("./AddCategory");

describe('Testing in AddCategory', () => {
    const setCategories = jest.fn();

    let wrapper = shallow(<AddCategory setCategories = {setCategories } /> )

    beforeEach ( () => {

        // limpiamos el wrapper xq tenia el inputValue en el primer caso
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories = {setCategories } /> )

    })

    test('should see Snapshot', async () => {
        expect(wrapper).toMatchSnapshot(); 
    });

    test('should  to change textbox Hola Mundo', async () => {
        const value = 'Hola'
        const input = wrapper.find(TextField)
        input.props().onChange( {target : { value }});
        const text = wrapper.find('p').text().trim();

        expect(text).toBe(value)
    });

    test('no debe de posterar la informacion con submit', async () => {

        wrapper.find('form').simulate('submit', {preventDefault(){}})
        expect ( setCategories ).not.toHaveBeenCalled()
        
    });

    test('llamar el setCategories y limpiar la caja de texto ', async () => {

        const value = 'Hola'
        const input = wrapper.find(TextField)
        input.props().onChange( {target : { value }});

        wrapper.find('form').simulate('submit', {preventDefault(){}})

        
        // evalua que la funcion se ejecute con una funcion =  setCategories(category => [inputValue, ...category, ]);

        expect ( setCategories ).toHaveBeenCalledWith( expect.any(Function))
        expect(input.props().value).toBe('') 
        
    });
});