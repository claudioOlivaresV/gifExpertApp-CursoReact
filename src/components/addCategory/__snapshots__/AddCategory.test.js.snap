// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in AddCategory should see Snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <WithStyles(ForwardRef(TextField))
    InputProps={
      Object {
        "endAdornment": <WithStyles(ForwardRef(InputAdornment))
          position="end"
        >
          <WithStyles(ForwardRef(IconButton))
            aria-label="toggle password visibility"
            onClick={[Function]}
          >
            <Memo />
          </WithStyles(ForwardRef(IconButton))>
        </WithStyles(ForwardRef(InputAdornment))>,
        "startAdornment": <WithStyles(ForwardRef(InputAdornment))
          position="start"
        >
          <Memo />
        </WithStyles(ForwardRef(InputAdornment))>,
      }
    }
    className=" my-3 px-2"
    fullWidth={true}
    id="standard-basic"
    label="Busar"
    onChange={[Function]}
    type="text"
    value=""
  />
  <p>
     
  </p>
</form>
`;
